src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs

```diff
@@ -76,7 +79,17 @@ public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeP
                     writer.Flush();
                 }
 #if NETCOREAPP
-                textWriter.Write(Encoding.UTF8.GetString(output.WrittenMemory.Span));
+                var messageBytes = output.WrittenMemory.Span;
+                var logMessageBuffer = ArrayPool<char>.Shared.Rent(Encoding.UTF8.GetCharCount(messageBytes));
```

Comment:
You probably want `GetMaxCharCount` here as `GetCharCount` is relatively expensive.

Reply:
`GetMaxCharCount` is a worst case, which will result in (over) allocating large buffers for long string.

Reply:
`GetCharCount` on the other hand is quite slow and takes almost as much as doing `GetChars` IIRC. Since we're pooling, overallocating probably shouldn't matter here.

Reply:
@MichalPetryka  Updated